### PR TITLE
Update ServerRequest to accept server name

### PR DESCRIPTION
**Pull request by MrAlias**
The `net.host.name` and `net.host.port` need to be set to the primary server name if it is known[^1]. Currently, the `ServerRequest` function provided by the `httpconv` packages do not include this in the processing of relevant attributes. This means for a user of these packages that know the primary server name, they need to manually added the attribute after a call to `ServerRequest`. This relies on the last-one-wins semantics of a span `SetAttributes` method which is not common knowledge, nor should it be needed for a user to correctly have their attributes defined.

Instead, update the `ServerRequest` method accept the `server` and use it to decide on how to construct the appropriate attributes.

[^1]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.17.0/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions

### CODE REVIEW

**Comment by dmathieu** [comment]
Isn't this a breaking change for the `semconv/vxxx/httpconv` packages? Shouldn't there be a changelog entry?

**Comment by MrAlias** [comment]
> Isn't this a breaking change for the `semconv/vxxx/httpconv` packages? Shouldn't there be a changelog entry?

This hasn't been released yet.

**Review by Aneurysm9 — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: internal/tools/semconvkit/templates/httpconv/http.go.tmpl [hunk 1]
```diff
@@ -88,13 +88,22 @@ func ClientStatus(code int) (codes.Code, string) {
 }
 
 // ServerRequest returns attributes for an HTTP request received by a server.
+//
+// The server must be the primary server name (the default virtual host if
```
[the remark this answers is not in the document]
Comment (MrAlias) [reply internal/tools/semconvkit/templates/httpconv/http.go.tmpl:92]: Yeah, that's fair. This uses the terms from the semantic conventions, but, to be honest, they themselves are using confusing terminology.

I'll update docs with what this means for Apache and Nginx.

[the remark this answers is not in the document]
Comment (MrAlias) [reply internal/tools/semconvkit/templates/httpconv/http.go.tmpl]: Updated, PTAL.

[the remark this answers is not in the document]
Comment (dashpole) [reply internal/tools/semconvkit/templates/httpconv/http.go.tmpl]: looks good, thanks.

**Pull request merged**